From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Sun, 28 May 2023 13:13:33 +0200
Subject: [PATCH] fix building with gcc 13

Without this building fails with:

ecperf.c: In function 'genericThread':
ecperf.c:99:24: error: storing the address of local variable 'sig' in '*threadData.p2' [-Werror=dangling-pointer=]
ecperf.c:91:13: note: 'sig' declared here
ecperf.c:86:21: note: 'data' declared here
ecperf.c: In function 'PKCS11Thread':
ecperf.c:71:24: error: storing the address of local variable 'sig' in '*threadData.p2' [-Werror=dangling-pointer=]
ecperf.c:53:13: note: 'sig' declared here
ecperf.c:47:20: note: 'data' declared here
cc1: all warnings being treated as errors

threadData is a pointer argument so any member variable that is set may
be used after the function ends. So in theory, setting threadData->p2
can be a problem.

It's not actually used after the end of the function so just clear it
again to make the compiler happy.

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 nss/cmd/ecperf/ecperf.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/nss/cmd/ecperf/ecperf.c b/nss/cmd/ecperf/ecperf.c
index 705d68f358f4..418e2b20c2e7 100644
--- a/nss/cmd/ecperf/ecperf.c
+++ b/nss/cmd/ecperf/ecperf.c
@@ -79,6 +79,8 @@ PKCS11Thread(void *data)
         }
         threadData->count++;
     }
+    if (threadData->isSign)
+        threadData->p2 = NULL;
     return;
 }
 
@@ -107,6 +109,8 @@ genericThread(void *data)
         }
         threadData->count++;
     }
+    if (threadData->isSign)
+        threadData->p2 = NULL;
     return;
 }
 
